pkg/codegen/dotnet/utilities.go

```diff
@@ -146,6 +146,16 @@ func getHelperMethodIfNeeded(functionName string, indent string) (string, bool)
 %s{
 %s    throw new System.NotImplementedException(errorMessage);
 %s}`, indent, indent, indent, indent), true
+	case "singleOrNone":
+		return fmt.Sprintf(`
+%sstatic T SingleOrNone<T>(T[] elements)
```

Comment:
I suggest changing `T[]` to `IEnumerable<T>` because `SingleOrNone` will be used a lot in with `ForExpressions` which compile to LINQ sequences resulting in `IEnumerable`s

Reply:
Fixed to just use LINQ's `Single` which has the exact behavior we want. Decided to not even generate a utility function, but to call that directly.